### PR TITLE
s2geometry: 0.9.0 → 0.11.1, enable on darwin

### PR DESCRIPTION
**Pull request by sikmir**
###### Description of changes
[Changelog](https://github.com/google/s2geometry/releases)

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
@ofborg build pure-maps

Fails for me on aarch64-linux

**Comment by sikmir** [comment]
Cross compilation for `aarch64-linux` works for me.

**Comment by risicle** [comment]
The ofborg build for `pure-maps` fails even for x86_64 linux.

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 203123` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>pure-maps</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>s2geometry</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -22249,7 +22249,9 @@ with pkgs;
 
   rustc-demangle = callPackage ../development/libraries/rustc-demangle { };
 
-  s2geometry = callPackage ../development/libraries/s2geometry { };
+  s2geometry = callPackage ../development/libraries/s2geometry {
+    abseil-cpp = abseil-cpp_202111;
```
Comment (Thra11) [inline pkgs/top-level/all-packages.nix:22253]: Change this to `abseil-cpp = abseil-cpp_202206` and it should build (Works for me at least).

**Review by Thra11 — CHANGES_REQUESTED** [review]
~Can't comment on darwin, but if it builds on darwin that's a step forward.
Otherwise, I think you just need to bump the abesil-cpp version and then this will be good to go!~
I was mistaken. There's still the issue with pure-maps not finding the abseil-cpp headers when inlcuding s2geometry's headers.

**Comment by Thra11** [comment]
@ofborg build pure-maps

**Comment by simonhollingshead** [comment]
Can you please see if v0.11.x fixes your issues?  #282245 requires C++14 or above now and this change was made in v0.11.0.

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 203123` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>pure-maps</li>
    <li>s2geometry</li>
  </ul>
</details>

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 203123` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>s2geometry</li>
  </ul>
</details>

**Comment by sikmir** [comment]
@risicle @Thra11 I've fixed the build.

**Review by FliegendeWurst — APPROVED** [review]

**Comment by sikmir** [comment]
https://github.com/NixOS/nixpkgs/pull/351558

**Pull request closed without merging**